prov/efa/src/efa_domain.c

```diff
@@ -325,11 +325,16 @@ int efa_domain_open(struct fid_fabric *fabric_fid, struct fi_info *info,
 			EFA_INFO(FI_LOG_DOMAIN, "EFA MR cache enabled, max_cnt: %zu max_size: %zu\n",
 			         cache_params.max_cnt, cache_params.max_size);
 			return 0;
+		} else {
```

Comment:
This is not what I meant.  I meant that the common case end of function should be obvious before you get into cleanup.  There's no need to jump into disable_caches if we never allocated a cache.  So make that an error state, and return 0 right before that.  THen remove the return 0 in line 327.